pennylane/templates/embeddings.py

```diff
@@ -228,28 +172,37 @@ def circuit(f=None):
 
     #############
     # Input checks
-    _check_no_variable([pad, normalize], ['pad', 'normalize'])
-    wires, n_wires = _check_wires(wires)
 
-    n_ampl = 2**n_wires
+    _check_no_variable(pad, msg="'pad' cannot be differentiable")
+    _check_no_variable(normalize, msg="'normalize' cannot be differentiable")
+
+    wires = _check_wires(wires)
+
+    n_amplitudes = 2**len(wires)
+    expected_shape = (n_amplitudes,)
     if pad is None:
-        msg = "AmplitudeEmbedding must get a feature vector of size 2**len(wires), which is {}. Use 'pad' " \
-               "argument for automated padding.".format(n_ampl)
-        shp = _check_shape(features, (n_ampl,), msg=msg)
+        shp = _check_shape(features, expected_shape, msg="'features' must be of shape {}; got {}. Use 'pad' "
+                                                         "argument for automated padding."
+                                                         "".format(expected_shape, _get_shape(features)))
     else:
-        msg = "AmplitudeEmbedding must get a feature vector of at least size 2**len(wires) = {}.".format(n_ampl)
-        shp = _check_shape(features, (n_ampl,), msg=msg, bound='max')
+        shp = _check_shape(features, expected_shape, bound='max', msg="'features' must be of shape {} or smaller "
+                                                                      "to be padded; got {}"
+                                                                      "".format(expected_shape, _get_shape(features)))
+
+    _check_type(pad, [float, complex, type(None)], msg="'pad' must be a float or complex; got {}".format(pad))
+    _check_type(normalize, [bool], msg="'normalize' must be a boolean; got {}".format(normalize))
 
-    _check_type(pad, [float, complex, type(None)])
-    _check_type(normalize, [bool])
     ###############
 
-    # Pad
+    #############
+    # Preprocessing
+
+    # pad
     n_feats = shp[0]
```

Comment:
I know I'm probably nitpicking way too much about the shortened variable names, but when you're already renaming stuff you could also rename `n_feats` to `n_features` and `shp` to `shape` -- for me feats is a regular word and shp reads as ship.